cs-commons/src/main/java/io/cloudslang/content/utils/BooleanUtilities.java

```diff
@@ -0,0 +1,61 @@
+package io.cloudslang.content.utils;
+
+import io.cloudslang.content.constants.BooleanValues;
+import io.cloudslang.content.constants.ExceptionsValues;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** A Boolean utility class that offers boolean string validation and conversion
+ * Created by victor on 31.08.2016.
+ */
+public final class BooleanUtilities {
+
+    private BooleanUtilities() {}
+
+    /** Given a string, it lowercase it and strips the blank spaces from the ends
+     * @param string the string to check
+     * @return the string in lowercase
+     */
+    @NotNull
+    private static String getLowerCaseString(@NotNull final String string) {
+        return StringUtils.strip(string).toLowerCase();
+    }
+
+    /** Given a boolean in string format, it checks if it's 'true' or 'false' (case insensitive)
+     * @param booleanStr the string to check
+     * @return true if booleanStr is 'true' or 'false' otherwise false
+     */
+    public static boolean isValid(@Nullable final String booleanStr) {
+        if (StringUtils.isBlank(booleanStr)) {
+            return false;
+        }
+        final String lowerCaseBoolean = getLowerCaseString(booleanStr);
+        return lowerCaseBoolean.equals(BooleanValues.TRUE) || lowerCaseBoolean.equals(BooleanValues.FALSE);
+    }
+
+    /** If the booleanStr is a valid boolean (see isValid) it returns the boolean value, otherwise it throws an exception
+     * @param booleanStr the string to convert to boolean
+     * @return true if the booleanStr is 'true', false if it's 'false'
+     * @throws IllegalArgumentException if the booleanStr is not a valid boolean
+     */
+    public static boolean toBoolean(@Nullable final String booleanStr) {
+        if (!isValid(booleanStr)) {
+            throw new IllegalArgumentException(booleanStr + ExceptionsValues.EXCEPTION_DELIMITER + ExceptionsValues.INVALID_BOOLEAN_VALUE);
```

Comment:
The case when 'abc' is passed as value and we need to parse to true or false a certain default exists in our content and at first look is not covered by the current utility methods.